pytest/tests/test_contract_update.py

```diff
@@ -0,0 +1,39 @@
+#/usr/bin/env python3
+"""
+Starts 2 near validators and 2 mpc nodes.
+Deploys v0 mpc contract.
+Proposes a contract update (v1).
+votes on the contract update.
+Verifies that the update was executed.
+"""
+
+import sys
+import time
+import pathlib
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[1]))
+from common_lib import shared, contracts
+
+
+def test_propose_update():
+    # deploy v0
+    contract_v0 = contracts.load_mpc_contract_v0()
+    cluster = shared.start_cluster_with_mpc(2, 2, 1, contract_v0)
+    # assert correct contract is deployed
+    cluster.assert_is_deployed(contract_v0)
+    # do some requests
+    cluster.send_and_await_signature_requests(2)
+    # propose v1
+    contract_v1 = contracts.load_mpc_contract_v1()
+    cluster.propose_update(contract_v1)
+    cluster.vote_update(2, 0)
```

Comment:
What happens if signature requests arrive while voting for an update is ongoing? Mainly want to be sure it doesn't break the upgrade process.

Reply:
That is fine. It will still be processed normally. The upgrade only happens when there are enough votes, which immediately sends a promise to deploy the contract code